src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.js

```diff
@@ -55,8 +57,48 @@ export const withReferenceValues = memo(props => {
     }
   };
 
+  const validateQuotedText = value => {
+    let isValid = true;
+    const pattern = /"[^"]+"/g;
+    const matches = value.match(pattern);
+
+    if (matches) {
+      const dataSet = dataOptions.map(option => option.name);
+
+      for (const str of matches) {
+        const croppedStr = str.slice(1, -1);
+
+        isValid = dataSet.some(data => data === croppedStr);
+
+        if (!isValid) {
+          return isValid;
```

Comment:
In this case, I think it’s better to use `break;` than `return isValid;`  Because in any case, you always return `isValid`, and this makes no sense to write `return` twice.